test-resources/headless-core-stub/lambdas/mock-jwks/src/services/cache-jwk.ts

```diff
@@ -0,0 +1,30 @@
+import { randomUUID } from "crypto";
+import { JSONWebKeySet, JWK } from "jose";
+import { getJwkKeyPair, GetJwkKeyPairOptions } from "../../../../utils/src/keypair";
+import { getHashedKid } from "../../../../utils/src/hashing";
+import { CORE_STUB_SIGNING_PUBLIC_JWK } from "../../../../utils/src/constants";
+
+let cachedKeys: { jwks: JSONWebKeySet } | null = null;
+
+export const generateJWKS = async (): Promise<{ jwks: JSONWebKeySet }> => {
+    if (cachedKeys) {
+        return cachedKeys;
+    }
+
+    const currentPublicKey = CORE_STUB_SIGNING_PUBLIC_JWK as JWK;
+
+    const [key1, key2] = await Promise.all([
+        getJwkKeyPair({ currentPublicKey } as GetJwkKeyPairOptions),
+        getJwkKeyPair({ kid: getHashedKid(randomUUID()) }),
```

Comment:
Q: If we are using a randomUUID() here for the kid, how will this work for the integration-tests? Or does the start endpoint use this? Just trying to understand the flow here in my head as in my ticket we will be doing: 

```
signedJWT.getHeader().getKeyID()
```
and using the value received from here to find the correct entry your JWKS endpoint.

Reply:
`getJwkKeyPair({ kid: getHashedKid(randomUUID()) })` belongs to the new public signing key, we use the first key which is the first position and is the one we currently have a private key in SSM

Reply:
Yeah the one with the random KID will never actually be used to sign a JWT so the session handler so should need to use it - it's just to replicate the conditions of a rotation where we have two keys returned (one used, and one unused)